### ISSUE TITLE
Feature Request: Enable the ability to change default site addresses on AT sites

### ISSUE THREAD

**Issue by cuemarie** [Automattic/wp-calypso#87140]
### What

Allow `.wpcomstaging.com` site addresses to be changed the way we allow `.wordpress.com` addresses.

### Why

Currently, `.wpcomstaging.com` domains are fixed and cannot be changed. Figuring out how to allow customers to update these would be a very developer-friendly improvement, as it can get confusing/disorganized when sites change and their default site addresses cannot be updated to match; anyone managing many sites will suffer more from this. 

Example: 7682267-zen

> I am splitting two sites - is there a way you can adjust the name of the wpcomstaging domain so it's [something matching the custom domain] instead of [the old sitename]? 
> yes... just organizationally... 
> i like consistency

### How

No idea – and I recognize there's probably some complications to allowing this, but thought it could be a nice improvement, especially for developers (re: p58i-gwE-p2 )

**Comment by cuemarie** [comment]
Another request here: 8247147-zen

**Comment by cuemarie** [comment]
Another customer request here: https://a8c.zendesk.com/agent/tickets/8702462

**Comment by cuemarie** [comment]
Some discussion relevant to this issue here: p1726668962420429-slack-C06PZC1RR5F

TLDR: When migrating sites in for customers, we frequently see default addresses that don't make sense (typically they were generated using the customer's username+number); its tempting to update those addresses for them to see if we can avoid the feature gap down the line if they want to update those addresses.

**Comment by mmtr** [comment]
This is still an issue.

On Simple sites, there is a "Change site address" that can be use to change `wordpress.com` domains:

<img width="1071" alt="Image" src="https://github.com/user-attachments/assets/d596a3e4-1b21-4ec5-ac46-7268f93ab6f8" />

<img width="1080" alt="Image" src="https://github.com/user-attachments/assets/616a6871-aa77-4245-b962-229fadd81572" />

But that option is not available on Atomic sites:

<img width="969" alt="Image" src="https://github.com/user-attachments/assets/c6b62848-7326-4cff-b9ab-5067212e4cdf" />

@Automattic/nomado is this something you can take a look at? Thanks!
